inflate.c

```diff
@@ -371,9 +371,18 @@ static int updatewindow(z_stream *strm, const unsigned char *end, uint32_t copy)
 
     /* if it hasn't been done already, allocate space for the window */
     if (state->window == NULL) {
+#ifdef INFFAST_CHUNKSIZE
+        unsigned wsize = 1U << state->wbits;
+        state->window = (unsigned char FAR *)
+                        ZALLOC(strm, wsize + INFFAST_CHUNKSIZE,
+                               sizeof(unsigned char));
```

Comment:
I also tried using `mmap()` to fold the window buffer around on itself to remove even more of these tests, but to show any real benefit it's necessary to tweak the behaviour so that inflate is performed directly into the window and then copied out when it's done.  Obviously this only suits callers that pass very small output buffers, but that's what libpng (a very performance-hostile caller) does.  Some platforms require that the window's padded to 64kB to do this, though.